mwclient/errors.py

```diff
@@ -22,6 +22,36 @@ def __init__(self, code, info, kwargs):
         super(APIError, self).__init__(code, info, kwargs)
 
 
+class NotFound(APIError):
+
+    def __init__(self, entity_type, code='missing', info=None, kwargs=None):
+        if not info:
+            info = 'Entity "%s" has not been found.' % entity_type
+        self.entity_type = entity_type
+        super(NotFound, self).__init__(code, info, kwargs)
+
+
+class CreateError(APIError):
+
+    def __init__(self, entity_type, code=None, info=None, kwargs=None):
+        if not info:
+            info = 'Entity "%s" can not been created.' % entity_type
+        self.entity_type = entity_type
+        super(CreateError, self).__init__(code, info, kwargs)
```

Comment:
I'm a bit uneasy about `CreateError` and `NotFound` as I don't really see the use for them. It's useful to be able to distinguish between *a specific* API error response (`UserCreateError`) and *any* API error response (`APIError`), but I don't see the need for one more level between those. Even if we create a new `PageCreateError` and makes it extend `CreateError`, I don't see the need to catch *some kind of creation failure* (either user or page).

Could you elaborate why you think these can be useful if you want to keep them?

Reply:
Mmm yes you right. Seemed to be a good idea when I added this to have a common Exception, allowing contributors to add future *NotFound subclasses.